addons/auth_signup/models/res_partner.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+import werkzeug
+
+from datetime import datetime, timedelta
+from urlparse import urljoin
+
+from openerp import api, fields, models, _
+
+from openerp.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT, ustr
+
+
+class SignupError(Exception):
```

Comment:
I see you have defined SignupError, random_token() and now() in both res_partner.py and res_users.py. You should define in any one of them and you should import it in another.